proposal.rst

```diff
@@ -168,27 +210,83 @@ Project Details
 
 **Describe any existing work and concepts on which your project is based.**
 
-FILLME
+My work will be based on the GSoC 2017 Project `Clickstream Mining for 
+Learning to Rank training data <https://trac.xapian.org/wiki/GSoC2017/LetorClickstream>`_. 
+
+**Phase 1: Implementing Click Models**
+
+The previous work has implemented the simplified DBN model for clickstream mining. 
+I plan to finish the EM algorithm in order to get a better performance.
+
+
+Besides DBN model, I will implement two more click models, the Dependent click Model 
+(DCM) and the User Browsing Model (UBM). Both two are also highly 
+effective to improve document ranking. 
+
+I will first work on an abstract class for all click models and then try to implement 
+each model according to relevant papers. 
+
+
+
+**Phase 2: Documentation**
+
+The second phase of my work will be a detailed documentation of each model I 
```

Comment:
This is a good idea, but I'd suggest thinking in terms of different documentation for different audiences. People using the clickstream -> letor system don't need to understand the deep detail of each model; they only really need some guidance on how to choose the best one for their situation. People working on the code (developers of future Xapian functionality in this area) will need a deeper understanding, which is either "internals" documentation, or (and this is often sufficient) documentation in the code itself which references papers, explains how particular algorithms are implemented for Xapian, and so forth.

Reply:
The people who don't need a deep understanding of the system actually split into two subgroups - there's people using omega -> clickstream -> letor, and developers wanting to integrate the clickstream -> letor by feeding it click logs from their own Xapian-based search (for example, a non-web based search front-end) but not wanting to try to modify any Xapian code.

One of the original ideas behind adding the clickstream logging to omega was that as well as being useful in itself, it would provide a working example of how to implement this, and the analysis part would be usable for click logs generated by things apart from Omega.

Reply:
I will think through to develop a method to satisfy different groups of users.